hybrid/src/main/java/jetbrains/jetpad/hybrid/BaseHybridSynchronizer.java

```diff
@@ -329,7 +332,11 @@ public boolean isSupported(ContentKind<?> kind) {
           @Override
           public <T> T get(ContentKind<T> kind) {
             if (kind == TOKENS_CONTENT) {
-              return (T) Collections.unmodifiableList(tokens);
+              List<Token> result = new ArrayList<>(copiedTokens.size());
```

Comment:
@katepol This protects the clipboard content from edits in pasted tokens. Please try returning not copied elements and `EditingTestCase.editStringAfterPaste()` will fail — instances pasted twice become linked.